internal/store/store.go

```diff
@@ -390,6 +420,17 @@ func (bs *BlockStore) PruneBlocks(height int64, state sm.State) (uint64, int64,
 	if err != nil {
 		return 0, -1, err
 	}
+	bs.blocksDeleted += int64(pruned)
+
+	if bs.compact && bs.blocksDeleted >= bs.compactionInterval {
+		// Error on compaction should not be reason to halt the chain
+		// TODO add logger to keep track of error
+		err = bs.db.Compact(nil, nil)
+		if err != nil && bs.logger != nil {
+			bs.logger.Error("Error compacting block store", "err", err)
+		}
+		bs.blocksDeleted = 0
```

Comment:
what is the reason to set `bs.blocksDeleted` to `0` here after compaction ? Is this a statement saying that compaction worked (didn't return an error) ? 

So if you store this value in the block store for future reference on another prune round, does this line `bs.blocksDeleted += int64(pruned)` makes sense ? If compaction is working and you reset `blocksDeleted` to 0, in your condition `if bs.compact && bs.blocksDeleted >= bs.compactionInterval` this will only trigger when there are more blocks pruned than the interval. I am thinking you want to keep track of pruned blocks in more than one pruning iteration no ?

Reply:
It will be reset to 0 only if we have pruned enough blocks that we can trigger compaction and then we can start counting again. Otherwise the value is preserved accross calls. `bs` is a pointer so the value remains, unlike with the state store where it would always be reset to 0 so I removed it.

Reply:
OK, in this case if you decide to refactor the code above and return an error instead of just logging, then you will also need to modify this logic a bit to account for the early return.

Reply:
I will reset this only if there is no error in compaction. Otherwise it stays the same.